pyresample/ewa/__init__.py

```diff
@@ -67,205 +67,12 @@
-        array provided then the returned tuple is simply the singe points integer
-        and single output grid array.
-    """
-    if isinstance(data_in, (tuple, list)):
-        # we can only support one data type per call at this time
-        assert(in_arr.dtype == data_in[0].dtype for in_arr in data_in[1:])
-    else:
-        # assume they gave us a single numpy array-like object
-        data_in = [data_in]
-
-    # need a list for replacing these arrays later
-    data_in = [np.ascontiguousarray(d) for d in data_in]
-    # determine a fill value if they didn't tell us what they have as a
-    # fill value in the numpy arrays
-    if "fill" is None:
-        if np.issubdtype(data_in[0].dtype, np.floating):
-            fill = np.nan
-        elif np.issubdtype(data_in[0].dtype, np.integer):
-            fill = -999
-        else:
-            raise ValueError(
-                "Unsupported input data type for EWA Resampling: {}".format(data_in[0].dtype))
-
-    convert_to_masked = False
-    for idx, in_arr in enumerate(data_in):
-        if isinstance(in_arr, np.ma.MaskedArray):
-            convert_to_masked = True
-            # convert masked arrays to single numpy arrays
-            data_in[idx] = in_arr.filled(fill)
-    data_in = tuple(data_in)
-
-    if out is not None:
-        # the user may have provided memmapped arrays or other array-like
-        # objects
-        if isinstance(out, (tuple, list)):
-            out = tuple(out)
-        else:
-            out = (out,)
-    else:
-        # create a place for output data to be written
-        out = tuple(np.empty(area_def.shape, dtype=in_arr.dtype)
-                    for in_arr in data_in)
-
-    # see if the user specified rows per scan
-    # otherwise, use the entire swath as one "scanline"
-    rows_per_scan = rows_per_scan or data_in[0].shape[0]
-
-    results = _fornav.fornav_wrapper(cols, rows, data_in, out,
-                                     np.nan, np.nan, rows_per_scan,
-                                     weight_count=weight_count,
-                                     weight_min=weight_min,
-                                     weight_distance_max=weight_distance_max,
-                                     weight_delta_max=weight_delta_max,
-                                     weight_sum_min=weight_sum_min,
-                                     maximum_weight_mode=maximum_weight_mode)
-
-    def _mask_helper(data, fill):
-        if np.isnan(fill):
-            return np.isnan(data)
-        else:
-            return data == fill
-
-    if convert_to_masked:
-        # they gave us masked arrays so give them masked arrays back
-        out = [np.ma.masked_where(_mask_helper(out_arr, fill), out_arr)
-               for out_arr in out]
-    if len(out) == 1:
-        # they only gave us one data array as input, so give them one back
-        out = out[0]
-        results = results[0]
-
-    return results, out
+from .ewa import ll2cr, fornav  # noqa
```

Comment:
Wouldn't using the absolute import `from pyresample.ewa import ll2cr, fornav` be better, and not require the `# noqa` directive?